airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java

```diff
@@ -183,17 +194,28 @@ public AirbyteMessageConsumer getConsumer(final JsonNode config,
    * @return use default AirbyteSchema or build using JsonSchema
    */
   protected boolean isDefaultAirbyteTmpTableSchema() {
-    return true;
+    return !denormalize;
   }
 
   protected Map<UploaderType, BigQueryRecordFormatter> getFormatterMap(final JsonNode jsonSchema) {
-    return Map.of(UploaderType.STANDARD, new DefaultBigQueryRecordFormatter(jsonSchema, getNamingResolver()),
-        UploaderType.CSV, new GcsCsvBigQueryRecordFormatter(jsonSchema, getNamingResolver()),
-        UploaderType.AVRO, new GcsAvroBigQueryRecordFormatter(jsonSchema, getNamingResolver()));
+    if (denormalize) {
+    return Map.of(UploaderType.STANDARD, new DefaultBigQueryDenormalizedRecordFormatter(jsonSchema, getNamingResolver()),
+        UploaderType.AVRO, new GcsBigQueryDenormalizedRecordFormatter(jsonSchema, getNamingResolver()));
+    } else {
+      return Map.of(UploaderType.STANDARD, new DefaultBigQueryRecordFormatter(jsonSchema, getNamingResolver()),
+              UploaderType.CSV, new GcsCsvBigQueryRecordFormatter(jsonSchema, getNamingResolver()),
+              UploaderType.AVRO, new GcsAvroBigQueryRecordFormatter(jsonSchema, getNamingResolver()));
+    }
   }
 
   protected String getTargetTableName(final String streamName) {
-    return namingResolver.getRawTableName(streamName);
+    if (denormalize) {
+    // This BigQuery destination does not write to a staging "raw" table but directly to a normalized
+    // table
+    return getNamingResolver().getIdentifier(streamName);
```

Comment:
we do we use different approaches :  getNamingResolver() and namingResolver itself?

Reply:
Yes, this is my mistake. Earlier in bigquery-denormalized we used getter. now, we can use namingResolver directions. Changed that.